src/meta/src/stream/mod.rs

```diff
@@ -51,14 +51,15 @@ pub fn record_table_vnode_mappings(
             hash_mapping_manager.set_fragment_state_table(fragment_id, node.table_id);
         }
         NodeBody::HashAgg(node) => {
-            let table_ids = node.get_table_ids();
-            for table_id in table_ids {
-                hash_mapping_manager.set_fragment_state_table(fragment_id, *table_id);
+            for table in &node.internal_tables {
+                hash_mapping_manager.set_fragment_state_table(fragment_id, table.id);
             }
         }
         NodeBody::HashJoin(node) => {
-            hash_mapping_manager.set_fragment_state_table(fragment_id, node.left_table_id);
-            hash_mapping_manager.set_fragment_state_table(fragment_id, node.right_table_id);
+            hash_mapping_manager
+                .set_fragment_state_table(fragment_id, node.left_table.clone().unwrap().id);
```

Comment:
```suggestion
                .set_fragment_state_table(fragment_id, node.left_table.as_ref().unwrap().id);
```
And there are other clones that could be avoided in this way.